src/http_server.cc

```diff
@@ -235,8 +236,24 @@ HTTPServer::Start()
 }
 
 TRITONSERVER_Error*
-HTTPServer::Stop()
+HTTPServer::Stop(uint32_t* exit_timeout_secs, const std::string& service_name)
 {
+  if (exit_timeout_secs != nullptr) {
```

Comment:
I think the general pattern would be something like:

```
stop_accepting_new_conn_ = true
if (exit_timeout_secs != nullptr) {
while (*exit_timeout_secs >0 && conn_cnt_>0) {
  sleep(1);
  *exit_timeout_secs --;
}
}
```
Note: not sure why we need to pass a pointer here for exit_timeout_secs - could we just set it to 0 if it wasn't set?
Feel like I'm missing something though that came up in testing. will try to sync offline

Reply:
updated to our conclusion synced offline

> not sure why we need to pass a pointer here for exit_timeout_secs - could we just set it to 0 if it wasn't set?

The purpose for the pointer is to pass the remaining exit_timeout_secs back to the caller, so it can be set in the core.

Both arguments are optional to remain compatible with the previous definition of `HTTPServer::Stop()`, and the exit_timeout_secs is default initialized to nullptr.